images/plbase/python-requirements.txt

```diff
@@ -1,47 +1,47 @@
 ansi2html==1.9.1
-black==23.12.1
+black==24.1.1
 chevron==0.14.0
 coloraide==2.16
-Faker==22.0.0
+Faker==22.6.0
 flake8-black==0.3.6
 flake8-isort==6.1.1
 Flake8-pyproject==1.2.3
-flake8==6.1.0
+flake8==7.0.0
 isort==5.13.2
 lxml==4.9.3
 lxml-stubs==0.4.0
 matplotlib==3.8.2
 networkx==3.2.1
 nltk==3.8.1
-numpy==1.26.2
+numpy==1.26.3
 openpyxl==3.1.2
 pandas-stubs==2.1.4.231227
 pandas==2.1.4
 Pint==0.23
-psutil==5.9.7
-PuLP==2.7.0
-pycryptodome==3.19.1
+psutil==5.9.8
+PuLP==2.8.0
+pycryptodome==3.20.0
 Pygments==2.17.2
 pygments-ansi-color==0.3.0
 pygraphviz==1.6
 pyquaternion==0.9.9
 pytest-cov==4.1.0
 pytest-lazy-fixture==0.6.3
 pytest-repeat==0.9.3
-pytest==7.4.4
+pytest==8.0.0
```

Comment:
This is the only major version bump to be a little wary about, but instructor code doesn't use it so probably not an issue.

Reply:
> instructor code doesn't use it

That's a bit of a dangerous assumption. I use pytest in a few of my questions (though in the autograder, not in server.py). That said, I agree that pytest use in plbase instructor code is expected to be uncommon.

Reply:
Also, I see the possibility that custom elements could be implemented with pytest checks in some scenarios for dev checks, though I don't foresee a case where the inconvenience of a possible (and likely uncommon) breaking change in dev work would be a reason not to update here.